### PR TITLE
build(pip): Enable the new 2020 resolver

### PR DESCRIPTION
**Pull request by BYK**
See https://app.asana.com/0/1143292751585859/1190293723047433/f and https://pip.pypa.io/en/latest/user_guide/#changes-to-the-pip-dependency-resolver-in-20-2-2020

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -116,7 +116,7 @@ RUN set -x \
   && mkdir -p $SENTRY_CONF
 
 COPY /dist/*.whl /tmp/dist/
-RUN pip install /tmp/dist/*.whl && pip check && rm -rf /tmp/dist
+RUN pip install /tmp/dist/*.whl --use-feature=2020-resolver && pip check && rm -rf /tmp/dist
```
Comment (joshuarli) [inline docker/Dockerfile:119]: Hmm, we don't upgrade pip in this dockerfile, it'll probably fail because this was introduced very recently.

Comment (joshuarli) [reply docker/Dockerfile:119]: Yeah: `no such option: --use-feature`

So I'd upgrade, pinned with a lower bound that has this cli kwarg. Also while you're at it, could we move the "sane defaults for pip" env vars to a separate ENV just before pip upgrade?

Comment (BYK) [reply docker/Dockerfile:119]: >could we move the "sane defaults for pip" env vars to a separate ENV just before pip upgrade?

We could but that would create another layer, which I don't want. Do you need this for something else?

>Yeah: no such option: --use-feature

Interesting, I expected these images to be recent enough. I guess not. I'll upgrade and see how it goes from there.

**Review by joshuarli — APPROVED** [review]
Assuming it works. (It should.)

**Comment by BYK** [comment]
>Assuming it works. (It should.)

Tests passed, which is all I care. You may wanna do the same for GHA/Travis since it works now?

**Comment by joshuarli** [comment]
So this broke when merged, no idea why, but looking into it.

**Comment by joshuarli** [comment]
I'm just going to revert for now, this is now failing everywhere.

**Pull request merged**